src/GraphQL/Types/Collections/SchemaTypes.cs

```diff
@@ -183,6 +185,53 @@ private void CheckSealed()
             return lookup;
         }
 
+        /// <summary>
+        /// Applies all delegates specified by the middleware builder to the schema.
+        /// <br/><br/>
+        /// When applying to the schema, modifies the resolver of each field of each graph type adding required behavior.
+        /// Therefore, as a rule, this method should be called only once during schema initialization.
+        /// </summary>
+        public void ApplyMiddleware(IFieldMiddlewareBuilder fieldMiddlewareBuilder, ISchema schema)
+        {
+            var transform = (fieldMiddlewareBuilder ?? throw new ArgumentNullException(nameof(fieldMiddlewareBuilder))).Build();
+
+            // allocation free optimization if no middlewares are defined
+            if (transform != null)
+            {
+                ApplyMiddleware(transform, schema);
+            }
+        }
+
+        /// <summary>
+        /// Applies the speicifed middleware transform delegate to the schema.
+        /// <br/><br/>
+        /// When applying to the schema, modifies the resolver of each field of each graph type adding required behavior.
+        /// Therefore, as a rule, this method should be called only once during schema initialization.
+        /// </summary>
+        public void ApplyMiddleware(Func<ISchema, FieldMiddlewareDelegate, FieldMiddlewareDelegate> transform, ISchema schema)
```

Comment:
Thus, technically we still give the simple way to apply middlewares event after creating a schema. I think it's not bad.

Reply:
Yeah, although you can't remove middleware, and SchemaTypes is supposed to be "sealed", and the schema is now "initialized".  I would have to say it actually should throw an exception, but I didn't add code to do that.  I'll leave it as-is.

Reply:
We _can_ write a feature to remove/add middlewares on-fly. `ApplyMiddleware` could return `IDisposable` and "track" applied resolvers. `Dispose` will eventually set stored values back. In this way, you can dynamically add and remove some functionality to all fields. Although the same can be achieved by adding the desired middleware once, and already inside it operate with some kind of on-off condition.